lib/osf-components/addon/components/registries/partial-registration-modal/manager/component.ts

```diff
@@ -0,0 +1,92 @@
+import { action } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import { task } from 'ember-concurrency';
+import DS from 'ember-data';
+import { layout } from 'ember-osf-web/decorators/component';
+import NodeModel from 'ember-osf-web/models/node';
+import defaultTo from 'ember-osf-web/utils/default-to';
+
+import { HierarchicalListManager } from 'osf-components/components/registries/hierarchical-list';
+import template from './template';
+
+@layout(template)
+export default class PartialRegistrationModalManagerComponent extends Component.extend({
+    loadAllChildNodes: task(function *(this: PartialRegistrationModalManagerComponent) {
+        let allChildNodesIncludingRoot = yield this.store.query('node', {
+            filter: {
+                root: this.rootNode.id,
+            },
+        });
+        allChildNodesIncludingRoot = allChildNodesIncludingRoot.toArray();
+        this.set('nodesIncludingRoot', allChildNodesIncludingRoot.slice());
+        this.set('selectedNodes', allChildNodesIncludingRoot.slice());
+    }),
+}) implements HierarchicalListManager {
+    @service store!: DS.Store;
+    nodesIncludingRoot: NodeModel[] = defaultTo(this.nodesIncludingRoot, []);
+    selectedNodes: NodeModel[] = defaultTo(this.selectedNodes, []);
```

Comment:
Since `nodesIncludingRoot` and `selectedNodes` are set in `loadAllChildNodes`, I assume they're not meant to be passed-in as args. Could you make that clearer?

We've been loosely following a convention of defining the "public" params at the top of the component class, with some sort of `// private` comment to clearly separate them from all the implementation details.